neomodel/core.py

```diff
@@ -22,18 +22,27 @@ def drop_constraints(quiet=True, stdout=None):
     if not stdout or stdout is None:
         stdout = sys.stdout
 
-    results, meta = db.cypher_query("CALL db.constraints()")
-    pattern = re.compile(':(.*) \).*\.(\w*)')
-    for constraint in results:
-        # Versions prior to 4.0 have a very different return format
-        if constraint[0].startswith('CONSTRAINT '):
-            db.cypher_query('DROP {!s}'.format(constraint[0]))
-            match = pattern.search(constraint[0])
+    # From Neo4j 4.3 on, the syntax to list constraints is 'SHOW CONSTRAINTS'
+    # For older versions, this returns a CypherSyntaxError, and the correct syntax is 'CALL db.constraints()'
+    try:
+        results, meta = db.cypher_query("SHOW CONSTRAINTS")
+    except CypherSyntaxError:
+        results, meta = db.cypher_query("CALL db.constraints()")
+
+    results_as_dict = [dict(zip(meta, row)) for row in results]
+    for constraint in results_as_dict:
+        # Versions prior to 4.3 returned a 'description' column
+        if "description" in constraint:
+            db.cypher_query('DROP {!s}'.format(constraint["description"]))
+
+            pattern = re.compile(':(.*) \).*\.(\w*)')
+            match = pattern.search(constraint["description"])
+            stdout.write(''' - Dropping unique constraint and index on label {0} with property {1}.\n'''.format(
```

Comment:
Would it be worth starting to drop the ` "".format() ` way of doing string interpolation in favour of f-strings?

I would suggest that we go ahead with f-strings from now on and make it a task to turn every string interpolation to an f-string for a future release.

Reply:
Until neomodel supports python 3.7, it is not the case. But sad =(

Reply:
@olegchigirin I do not understand this comment, but there is nothing stopping us from making this change in an upcoming major release.

Reply:
Sorry, I was misleading.

f-string added in python 3.6, not python 3.8.

Yes, in this case f-string are useful staff to add

Reply:
@olegchigirin No worries, thank you for the clarification.

Reply:
Yes I agree, f-strings are better. And we can already do it because the current version of neomodel supports Python 3.7 -> 3.10

Reply:
Updated in upcoming commit